src/Helpers/ElementHelpers/ElementHelper.cs

```diff
@@ -203,49 +203,71 @@ public List<PageSummary> GenerateQuestionAndAnswersList(string guid, FormSchema
             {
+                var formSchemaQuestions = new List<IElement>();
+
+                if (page.Elements.Any(_ => _.Type == EElementType.AddAnother))
+                {
+                    var listOfPageSummary = new List<PageSummary>();
+                    var addAnotherElement = page.Elements.FirstOrDefault(_ => _.Type == EElementType.AddAnother);
+                    var currentIncrement = GetAddAnotherNumberOfFieldsets(addAnotherElement, formAnswers);
+                    for (var i = 1; i <= currentIncrement; i++)
+                    {
+                        var addAnotherPageSummary = new PageSummary
+                        {
+                            PageTitle = addAnotherElement.GetLabelText(page.Title),
+                            PageSlug = $"{page.PageSlug}-{addAnotherElement.Properties.QuestionId}-{i}"
+                        };
+
+                        var listOfNestedElements = page.ValidatableElements.Where(_ => _.Properties.QuestionId.Contains($":{i}:")).ToList();
+                        addAnotherPageSummary.Answers = GenerateSummaryAnswers(listOfNestedElements, page, formAnswers, false);
+                        listOfPageSummary.Add(addAnotherPageSummary);
+                    }
+                    
+                    formSummary.AddRange(listOfPageSummary);
+                }
+
                 var pageSummary = new PageSummary
                 {
                     PageTitle = page.Title,
                     PageSlug = page.PageSlug
                 };
 
-                var formSchemaQuestions = page.ValidatableElements
-                    .Where(_ => _ != null)
-                    .ToList();
+                formSchemaQuestions = page.ValidatableElements
+                .Where(_ => _ != null)
+                .ToList();
 
                 if (!formSchemaQuestions.Any() || !reducedAnswers.Where(p => p.PageSlug == page.PageSlug).Select(p => p).Any())
                     continue;
 
-                pageSummary.Answers = GenerateSummaryAnswers(formSchemaQuestions, page, formAnswers);
+                pageSummary.Answers = GenerateSummaryAnswers(formSchemaQuestions, page, formAnswers, true);
                 formSummary.Add(pageSummary);
             }
 
             return formSummary;
         }
 
-        public Dictionary<string, string> GenerateSummaryAnswers(List<IElement> formSchemaQuestions, Page page, FormAnswers formAnswers)
+        public int GetAddAnotherNumberOfFieldsets(IElement addAnotherElement, FormAnswers formAnswers)
+        {
+            var formDataIncrementKey = $"addAnotherFieldset-{addAnotherElement.Properties.QuestionId}";
+            return formAnswers.FormData.ContainsKey(formDataIncrementKey) 
+                ? int.Parse(formAnswers.FormData.GetValueOrDefault(formDataIncrementKey).ToString()) 
+                : throw new ApplicationException($"ElementHelper::GetCurrentAddAnotherIncrement, FormData key not found for {formDataIncrementKey}");

```

Comment:
would this throw if only a single item is added and you proceeded straight away?

Reply:
It shouldn't because the FormData gets added in ProcessAddAnother on submit